lib/blpapi-wrapper.ts

```diff
@@ -324,7 +334,11 @@ export class Session extends events.EventEmitter {
                 throw new Error('Unknown request type ' + requestName);
             }
             var responseEventName = REQUEST_TO_RESPONSE_MAP[requestName];
-            this.session.request(uri, requestName, request, correlatorId);
+            if (isAuthRequest) {
```

Comment:
I had callbacks before, and since there's exactly two types of requests in the underlying API, I switched to this form. I think this needs to be a shared function because of the difference is literally in the request type but am open to switching back to the callback form if you think that's really cleaner.